crates/red_knot/tests/cli.rs

```diff
@@ -199,14 +242,26 @@ fn configuration_rule_severity() -> anyhow::Result<()> {
     "#,
     )?;
 
-    assert_cmd_snapshot!(case.command(), @r"
-        success: false
-        exit_code: 1
-        ----- stdout -----
-        warning[lint:division-by-zero] <temp_dir>/test.py:2:5 Cannot divide object of type `Literal[4]` by zero
+    assert_cmd_snapshot!(case.command(), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    warning: lint:division-by-zero
+     --> <temp_dir>/test.py:2:5
+      |
+    1 |
+    2 | y = 4 / 0
+      |     ----- Cannot divide object of type `Literal[4]` by zero
```

Comment:
Ok really dumb question, but why does annotate-snippets choose `^^^^` in some cases but `----` here?

Reply:
Good question actually! It chooses a different mark based on the severity level:

https://github.com/astral-sh/ruff/blob/b0b8b062410afdee0c9948a23fc99de8e5a406b6/crates/ruff_annotate_snippets/src/renderer/display_list.rs#L716-L723

Honestly kinda meh on this myself since it's somewhat non-obvious to me.

Reply:
I think it makes more sense once you have e.g. (multiple) "Info" and "Error" annotations in a single diagnostic. Example from my own project (thinking about it, the function parameter annotation on top here should probably also be "info", but anyway):

![image](https://github.com/user-attachments/assets/ba8d4149-1d66-464b-a276-a43712c09b08)

I kind of like how the informal parts are just underlined with `---` and the actual error is underlined with `^^^` as if to say: look, this is what you need to change.

Reply:
Ah yeah that's a good point to consider too. I'll leave it as-is for now. This was just meant to get some decent rendering in. We can absolutely iterate on the exact format and what not.